lisa/sut_orchestrator/azure/features.py

```diff
@@ -2204,11 +2207,13 @@ def create_or_update(
 
         if protected_settings:
             add_secret(
-                str(extension_parameters.as_dict()["protected_settings"]),
+                str(extension_parameters.protected_settings),
```

Comment:
Like this change is not related to azure upgrading, and potentially regression. Make sure you have exact tool versions.